### DISCUSSION TITLE
Powerwall 3 APIs

### DISCUSSION THREAD

**Discussion by jasonacox** [vloschiavo/powerwall2#99] [General]
It seems that Tesla Powerwall 3 systems are starting to be installed. There is a webserver listening, but there doesn't seem to be a customer portal and the APIs seem to be missing or different. 

See https://github.com/jasonacox/Powerwall-Dashboard/issues/387

Does anyone have any details on these new PW3 systems and their APIs?

**Comment by vls29** [comment]
Interesting, thanks for the issue link.

Can't really suggest anything to help, other than it would be interesting to see what's being returned with a verbose curl (or browser tools network tab) on that IP address? Only because if I curl my PW2 on an invalid endpoint (e.g. `curl -v -k https://192.168.0.57/blah`), I get a 200 status code with some json + javascript and then it displays a bit of text saying "404 page not found".  I don't think it will help though as the video from jesaf00 doesn't appear to show styled 404 text, but maybe the headers in the response might give something away?

**Reply by jasonacox** [reply]
Good suggestion, @vls29 !

**Reply by jesaf00** [reply]
I pasted an output here: https://github.com/jasonacox/Powerwall-Dashboard/issues/387
